### DISCUSSION TITLE
notes identifed by "Query based FileClasses" can't be added a fileClass based on the plugin

### DISCUSSION THREAD

**Discussion by nocona71** [mdelobelle/metadatamenu#589] [Tips and Tricks]
Thanks @mdelobelle, I appreciate this plugin. I really like this unprecedented object oriented approach to structure a repo. I am looking forward to the updates which are hopfully following. 

At the moment I struggle at the following issue:

### Observed behaviour
When I "implicitly" identify a group of files instead of "explicitly"adding a fileClass to the corresponding note they are indicated as an "object" of that fileClass "project" indicated by the icon after the filename in the navigation. Here all files below the folder "project" shall be from fileClass "project":

#### Configuration Query based FileClasses based ton the folder
![image](https://github.com/mdelobelle/metadatamenu/assets/63201084/05e7813d-c8a5-4ca6-81f3-dd663376dc5d)

#### Example project which placed in the folder project
![image](https://github.com/mdelobelle/metadatamenu/assets/63201084/94e6906c-f1c8-4229-8b03-d9554c111269)

#### Trying to add an explicit fileClass to the note
![image](https://github.com/mdelobelle/metadatamenu/assets/63201084/a5955487-1815-4857-9923-1114b0e29055)

the modal doesn't offer to add the fileClass "project"

![image](https://github.com/mdelobelle/metadatamenu/assets/63201084/71362a84-9df9-47ad-9930-ee8135c9ecea)

#### optimal solution
Provision of an option to propagate/convert an "implicit" fileClass assigements (here identified by a dv query) to "explicit" fileClasses (e.g. by an additonal entry in the conetxt menu). Here a UI suggestion for a corresponding addition to the overview page of a "project" fileClass. file found as an implicit object of a fileClass can be propagated by using the "make explicit" button:

![image](https://github.com/mdelobelle/metadatamenu/assets/63201084/cc0b3c78-d1fa-4dbe-b362-69bcec3f310e)

Optional: button "make all explicit" to propagate all at once

Additional advantage: a combined view of implicitly and explicitly found notes belonging to one fileClass

#### easiest solution
Provision of the option to manually add the fileClass "project":
![image](https://github.com/mdelobelle/metadatamenu/assets/63201084/81e4440f-0a29-476e-8ec0-6beca373c9ab)

**Comment by Muved** [comment]
I'm having the same issue.

**Comment by mdelobelle** [comment]
This is a very niche edge case, so I won't be able to develop a specific feature for that.
Though you can achieve what you want with a simple script (that you can run from the console or from a templater script).
Here's a proposal:
```javascript
for(const p of dv.pages('"projects"')){ // replace dv.pages('"projects"') by any dvjs query to target your files
    MetadataMenu.api.postValues(
        p.file.path, 
        [
            {
                indexedPath: "fileclass-field-fileclass", // replace the trailing "fileclass" keyword by your fileclass alias if needed
                payload:    {value: "project"} // replace "project" by the value of your choice
            }
        ], 
        -1 // this will insert the fileclass field in the frontmatter
    )
}

```
